cpu/native/fs/native_fs.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_fs_native
+ * @{
+ *
+ * @file
+ * @brief       native integration with vfs
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "mutex.h"
+#include "native_internal.h"
+#include "fs/native_fs.h"
+#include "vfs.h"
+
+#define ENABLE_DEBUG 0
+#include <debug.h>
+
+/**
+ * @brief Assign each native instance a sub-sirectory based on `_native_id`
+ */
+#ifndef CONFIG_NATIVE_ISOLATE_FS
+#define CONFIG_NATIVE_ISOLATE_FS 0
+#endif
+
+#define FD(filep) filp->private_data.value
+#define DIRP(dirp) dirp->private_data.ptr
+
+/* Reentrancy guard required by the various static locals of the implementation */
+static mutex_t _lock;
+
+static void _do_prefix(vfs_mount_t *mountp, const char *name, char *buffer, size_t len)
+{
+    const native_desc_t *fs_desc = mountp->private_data;
+    size_t res;
+
+    if (getcwd(buffer, len) == NULL) {
+        puts("fs_native: can't get working directory");
+        exit(errno);
+    }
+
+    len = sizeof(buffer) - strlen(buffer);
+    assert(len > strlen(fs_desc->hostpath));
+
+    if (CONFIG_NATIVE_ISOLATE_FS) {
+        res = snprintf(&buffer[strlen(buffer)], len, "%s/%u/%s", fs_desc->hostpath, _native_id, name);
```

Comment:
```suggestion
        res = snprintf(&buffer[strlen(buffer)], len, "%s/%u%s", fs_desc->hostpath, _native_id, name);
```

for consistency with below where we also implicitly assume that name starts with a slash (an assumption we may want to make explicit, but not in here because the empty string is used by one caller)